plugins/portal-history-ornaments.js

```diff
@@ -0,0 +1,244 @@
+<div>
+<select id="portal-history-settings--display-mode">
+  <option value="received" ${portalsHistory.settings.drawMissing?'':'selected'}>Show uniques received</option>
+  <option value="missing" ${portalsHistory.settings.drawMissing?'selected':''}>Show missing uniques</option>
+</select>
+</div>
+<div><label style="color:red;"><input type="checkbox" id="portal-history-settings--show-visited" 
+  ${portalsHistory.settings.showVisitedCaptured?'checked':''}> Show visited/captured</label></div>
+<div><label style="color:violet;"><input type="checkbox" id="portal-history-settings--show-scouted" 
+  ${portalsHistory.settings.showScouted?'checked':''}> Show Scout Controlled</label></div>
+</div>`,
+
+    title: 'Portal History Settings',
+    id: 'plugin-portal-history-flags',
+    width: 'auto',
+    closeCallback: function () {
+      var elMode = document.getElementById('portal-history-settings--display-mode');
+      var elVisitedCaptured = document.getElementById('portal-history-settings--show-visited');
+      var elScouted = document.getElementById('portal-history-settings--show-scouted');
+
+      portalsHistory.settings.drawMissing = elMode.value === 'missing';
+      portalsHistory.settings.showVisitedCaptured = elVisitedCaptured.checked;
+      portalsHistory.settings.showScouted = elScouted.checked;
+
+      localStorage[KEY_SETTINGS] = JSON.stringify(portalsHistory.settings);
+      portalsHistory.drawAllFlags();
+    }
+  });
+}
+  
+function createIcons () {
+//  var LEVEL_TO_RADIUS = [7, 7, 7, 7, 8, 8, 9, 10, 11];
+  var LEVEL_TO_RADIUS =   [6, 6, 6, 6, 8, 8, 8, 10, 11];
+  var scale = window.portalMarkerScale();
+  portalsHistory.iconSemiMarked = {};
+  portalsHistory.iconMarked = {};
+	portalsHistory.iconScouted = {};
+  var parts = (portalsHistory.settings.showVisitedCaptured + portalsHistory.settings.showScouted);
+  LEVEL_TO_RADIUS.forEach((el, idx) => {
+    var size = (el * 2 + 8) * scale;
+    var offset = 0;
+    if (portalsHistory.settings.showScouted) {
+      portalsHistory.iconScouted[idx] = svgToIcon(getSVGString(size, 'violet', parts, offset), size + 4);
+    offset++;
+      } else {
+      portalsHistory.iconScouted[idx] = svgToIcon(getSVGString(size, 'transparent', parts, offset), size + 4);
+    }
+
+    if (portalsHistory.settings.showVisitedCaptured) {
+      portalsHistory.iconSemiMarked[idx] = svgToIcon(getSVGString(size, 'yellow', parts, offset), size + 4);
+      portalsHistory.iconMarked[idx] = svgToIcon(getSVGString(size, 'red', parts, offset), size + 4);
+      offset++;
+    } else {
+      portalsHistory.iconSemiMarked[idx] = svgToIcon(getSVGString(size, 'transparent', parts, offset), size + 4);
+      portalsHistory.iconMarked[idx] = svgToIcon(getSVGString(size, 'transparent', parts, offset), size + 4);
+    }
+  });
+}
+
+function drawPortalFlags (portal) {
+  var drawMissing = portalsHistory.settings.drawMissing;
+  portal._historyLayer = new L.LayerGroup();
+
+  if (portal.options.data.history) {
```

Comment:
```suggestion
  var history = portal.options.data.history;
  if (history) {
```
And then shorten the rest of `portal.options.data.history` occurences.

Reply:
done